src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -1039,6 +1085,21 @@ object TrieMap extends MapFactory[TrieMap] {
   class MangledHashing[K] extends Hashing[K] {
     def hash(k: K): Int = scala.util.hashing.byteswap32(k.##)
   }
+
+  private[concurrent] sealed trait RemovalPolicy {
+    def shouldRemove[V](a: V, b: V): Boolean
+  }
+  private[concurrent] object RemovalPolicy {
+    object Always extends RemovalPolicy {
+      def shouldRemove[V](a: V, b: V): Boolean = true
+    }
+    object FullEquals extends RemovalPolicy {
+      def shouldRemove[V](a: V, b: V): Boolean = a == b
+    }
+    object ReferenceEq extends RemovalPolicy {
+      def shouldRemove[V](a: V, b: V): Boolean = a.asInstanceOf[AnyRef] eq b.asInstanceOf[AnyRef]
+    }
+  }
```

Comment:
I would think the `Int` constant version would be shorter and probably faster than this.

Reply:
not shorter in the places where it counts though. in the actual use site, you'll end up with a long and complex conditional something like `removal == Always || (if (removal == FullEquals) v == v0 else v.asInstanceOf[AnyRef] eq v0.asInstanceOf[AnyRef])` (and that ignores that the constants will probably be prefixed for context)

Reply:
But the int constant version would be something like `test(removal, v, v0)`.  You wouldn't write out all that logic at the test site.  This is the kind of thing that inlines really well.

Reply:
I guess `def shouldRemove`, to maintain the lingo, rather than `def test`.  Point is, the logic doesn't change; it's just a function with an int argument instead of an overloaded method on a trait.

Reply:
done